python/main_loop.py

```diff
@@ -205,6 +231,11 @@ def createNewLocalPath(sailbot, maxAllowableRuntimeSeconds, desiredHeadingPublis
                 else:
                     rospy.loginfo("Keeping old local path")
 
+            elif ((hasUpwindOrDownwindOnPath or costTooHigh)
+                  and (createNewLocalPathAttempts <= CREATE_NEW_LOCAL_PATH_ATTEPT_THRESH)):
+                # Increase the counter if we did NOT update local path, but there WAS an "unimportant" reason to update.
+                createNewLocalPathAttempts += 1
+
```

Comment:
The only time we want to increment `createNewLocalPathAttempts` is when there is at least one reason to update the local path, none of the reasons(s) are "important", and so the attempt to create a new local path was blocked by `lowWindPermitsUpdate`.

If we end up checking the `elif` condition, it means that `mustUpdateLocalPath` was false and that `generateNewLocalPathToCompare` was also false. This already means that there were no "important" reasons to update, and that any "unimportant" reasons that may exist were blocked by `lowWindPermitsUpdate`. The last check we need to do before incrementing `createNewLocalPathAttempts` is to make sure that there WAS an "unimportant" reason, otherwise there was no actual attempt that got blocked.

I also made `createNewLocalPathAttempts` stop updating once it gets big enough, so we don't end up storing a large number if we go a long time before `createNewLocalPathAttempts` resets.

Reply:
> I also made createNewLocalPathAttempts stop updating once it gets big enough, so we don't end up storing a large number if we go a long time before createNewLocalPathAttempts resets.

Good thinking